pkg/eval/json_evaluator.go

```diff
@@ -103,15 +103,25 @@ func (je *JSONEvaluator) ResolveStringValue(flagKey string, context *structpb.St
 	return resolve[string](flagKey, context, je.evaluateVariant, je.state.Flags[flagKey].Variants)
 }
 
-func (je *JSONEvaluator) ResolveNumberValue(flagKey string, context *structpb.Struct) (
-	value float32,
+func (je *JSONEvaluator) ResolveFloatValue(flagKey string, context *structpb.Struct) (
+	value float64,
+	variant string,
+	reason string,
+	err error,
+) {
+	value, variant, reason, err = resolve[float64](flagKey, context, je.evaluateVariant, je.state.Flags[flagKey].Variants)
+	return
+}
+
+func (je *JSONEvaluator) ResolveIntValue(flagKey string, context *structpb.Struct) (
+	value int64,
 	variant string,
 	reason string,
 	err error,
 ) {
 	var val float64
 	val, variant, reason, err = resolve[float64](flagKey, context, je.evaluateVariant, je.state.Flags[flagKey].Variants)
-	value = float32(val)
+	value = int64(val)
```

Comment:
Is this safe? Why are we going to int64 via float64?

Reply:
Its my understanding that the returned number values from the json flag definitions are all of type float64, so the cast into int64 is required. (Although I would love to be wrong on this one as I'm not happy with the approach either).

As for the type conversion, technically it is unsafe as the conversion may not be lossless, however in these cases the float64 version should be used anyway - what are your thoughts?